api/management/api.go

```diff
@@ -187,4 +191,154 @@ func ConvertToKCUser(user UserRepresentation) kc.UserRepresentation {
 	}
 
 	return userRep
-}
\ No newline at end of file
+}
+
+// Validators
+
+func (user UserRepresentation) Validate() error {
+	if user.Id != nil && !matchesRegExp(*user.Id, RegExpID) {
```

Comment:
In another project, I added a nilAllowed boolean to the validate function and passed it to matchesRegExp. Then check for nil is done in matchesRegExp and if value is nil, I return nilAllowed. This way, I can Validate for creation (all fields might be valid and not nil) or for update (nil allowed)